SkyDrive/SkyDrive.BLL/Services/InstructorService.cs

```diff
@@ -0,0 +1,65 @@
+﻿using Microsoft.EntityFrameworkCore;
+using SkyDrive.BLL.Entities;
+using SkyDrive.BLL.Interfaces;
+using SkyDrive.Exceptions;
+
+namespace SkyDrive.BLL.Services
+{
+    public class InstructorService : IInstructorService
+    {
+        private readonly ApplicationContext _context;
+
+        public InstructorService(ApplicationContext context)
+        {
+            _context = context;
+        }
+
+        public async Task<IEnumerable<InstructorEntity>> GetAllInstructorEntities()
+        {
+            return await _context.Instructors.AsNoTracking().ToListAsync();
+        }
+
+        public async Task<InstructorEntity> GetInstructorEntityById(int id)
+        {
+            return await GetEntity(id);
+        }
+
+        public async Task<InstructorEntity> CreateInstructorEntity(InstructorEntity instructorEntity)
+        {
+            _context.Instructors.Add(instructorEntity);
+            await _context.SaveChangesAsync();
+
+            return instructorEntity;
+        }
+
+        public async Task<InstructorEntity> UpdateInstructorEntity(InstructorEntity instructorEntity)
+        {
+            var entity = await GetEntity(instructorEntity.Id);
+
+            _context.Instructors.Update(entity);
+            await _context.SaveChangesAsync();
+
```

Comment:
Looks like it will not work correctly, you get a changedEntity, then get an entity from the storage and provide it to the Update method, but all changed data stay inside changedEntity. The same poitnt for other services.